test/commons/text/is-icon-ligature.js

```diff
@@ -0,0 +1,191 @@
+		}
+
+		var materialFont = new FontFace(
+			'Material Icons',
+			'url(https://fonts.gstatic.com/s/materialicons/v48/flUhRq6tzZclQEJ-Vdg-IuiaDsNcIhQ8tQ.woff2)'
+		);
+		var robotoFont = new FontFace(
+			'Roboto',
+			'url(https://fonts.gstatic.com/s/roboto/v20/KFOmCnqEu92Fr1Mu4mxKKTU1Kg.woff2)'
+		);
+		window.Promise.all([materialFont.load(), robotoFont.load()]).then(
+			function() {
+				document.fonts.add(materialFont);
+				document.fonts.add(robotoFont);
+				done();
+			}
+		);
+	});
+
+	it('should return false for normal text', function() {
+		var target = queryFixture('<div id="target">Normal text</div>');
+		assert.isFalse(isIconLigature(target.children[0]));
+	});
+
+	it('should return false for emoji', function() {
+		var target = queryFixture('<div id="target">🌎</div>');
+		assert.isFalse(isIconLigature(target.children[0]));
+	});
+
+	it('should return false for non-bmp unicode', function() {
+		var target = queryFixture('<div id="target">◓</div>');
+		assert.isFalse(isIconLigature(target.children[0]));
+	});
+
+	it('should return false for whitespace strings', function() {
+		var target = queryFixture('<div id="target">     </div>');
+		assert.isFalse(isIconLigature(target.children[0]));
+	});
+
+	(fontApiSupport ? it : it.skip)(
+		'should return false for common ligatures (fi)',
+		function() {
+			var target = queryFixture(
+				'<div id="target" style="font-family: Roboto">figure</div>'
+			);
+			assert.isFalse(isIconLigature(target.children[0]));
+		}
+	);
+
+	(fontApiSupport ? it : it.skip)(
+		'should return false for common ligatures (ff)',
+		function() {
+			var target = queryFixture(
+				'<div id="target" style="font-family: Roboto">ffugative</div>'
+			);
+			assert.isFalse(isIconLigature(target.children[0]));
+		}
+	);
+
+	(fontApiSupport ? it : it.skip)(
+		'should return false for common ligatures (fl)',
+		function() {
+			var target = queryFixture(
+				'<div id="target" style="font-family: Roboto">flu shot</div>'
+			);
+			assert.isFalse(isIconLigature(target.children[0]));
+		}
+	);
+
+	(fontApiSupport ? it : it.skip)(
+		'should return false for common ligatures (ffi)',
+		function() {
+			var target = queryFixture(
+				'<div id="target" style="font-family: Roboto">ffigure</div>'
+			);
+			assert.isFalse(isIconLigature(target.children[0]));
+		}
+	);
+
+	(fontApiSupport ? it : it.skip)(
+		'should return false for common ligatures (ffl)',
+		function() {
+			var target = queryFixture(
+				'<div id="target" style="font-family: Roboto">fflu shot</div>'
+			);
+			assert.isFalse(isIconLigature(target.children[0]));
+		}
+	);
+
+	(fontApiSupport ? it : it.skip)(
+		'should return true for an icon ligature',
+		function() {
+			var target = queryFixture(
+				'<div id="target" style="font-family: \'Material Icons\'">delete</div>'
+			);
+			assert.isTrue(isIconLigature(target.children[0]));
+		}
+	);
+
+	(fontApiSupport ? it : it.skip)(
+		'should return true after the 3rd time the font is an icon',
```

Comment:
"3rd time" is arbitrary? It's not explained. I'm not sure that works for all cases. Is it possible to have a "normal" font with added ligatures?

Reply:
The use cases we are optimizing for:
- a non-icon ligature font (95% of cases)
- an exclusive icon ligature font (4.999%)
- a mix of regular text and icon ligatures (should be extremely rare. Material Icon doesn't even have data to draw latin characters)

For the rare case, we have [no choice but to check it every time](https://github.com/dequelabs/axe-core/pull/1829/files#diff-d286a2c961c899ecddcb0ee4aaf2ef48R124-R127). For the other two, we can identify them quickly (using the arbitrary 3 times seen as chosen by Wilco). If we encounter problems we can tweek things when we know about them.